lib/views/frontend/spree/checkout/payment/v3/_form_elements.html.erb

```diff
@@ -1,4 +1,4 @@
-<div id="payment-request-button" data-stripe-config="<%= payment_method.stripe_config(current_order).to_json %>"></div>
+<div id="payment-request-button" data-stripe-config="<%= payment_method.stripe_config(current_order).to_json %>" data-v3-api="<%= stripe_v3_api %>"></div>
```

Comment:
Do we always show this button in v3, or it should be wrapped in an if somehow? Or is it already done via JS?

Reply:
Currently, there's one flag to disable the button: 

```ruby
def payment_request?
  v3_intents? && preferred_stripe_country.present?
end
```
when this method returns false, the button is not shown. So, the preference `stripe_country` controls the button visibility, as it's a requirement for the payment request to work.

When the preference is not present, the JS code eventually will not initialize the button:

```js
    setUpPaymentRequest: function(opts) {
      var opts = opts || {};
      var config = this.config.payment_request;

      if (config) {
       // ....
```

It's probably worth documenting a bit this as well.